utility/src/main/java/org/oppia/android/util/networking/DebugNetworkConnectionUtil.kt

```diff
@@ -0,0 +1,45 @@
+package org.oppia.android.util.networking
+
+import org.oppia.android.util.networking.NetworkConnectionUtil.ConnectionStatus
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/**
+ * [NetworkConnectionUtil] that gets and sets the current [ConnectionStatus] of the device in debug builds.
+ */
+@Singleton
+class DebugNetworkConnectionUtil @Inject constructor(
+  private val prodNetworkConnectionUtil: ProdNetworkConnectionUtil
+) : NetworkConnectionUtil {
+
+  private var forcedConnectionStatus: ConnectionStatus? = null
+
+  override fun getCurrentConnectionStatus(): ConnectionStatus {
+    val actualConnectionStatus = prodNetworkConnectionUtil.getCurrentConnectionStatus()
+    if (actualConnectionStatus == ConnectionStatus.NONE) {
+      forcedConnectionStatus = null
+    }
+    forcedConnectionStatus?.let {
+      return it
+    }
+    return actualConnectionStatus
+  }
+
+  /**
+   * Forces [connectionStatus] as the current connection status of the device and returns a
+   * [Boolean] indicating result.
+   */
+  fun setCurrentConnectionStatus(connectionStatus: ConnectionStatus): Boolean {
+    if (prodNetworkConnectionUtil.getCurrentConnectionStatus() == ConnectionStatus.NONE &&
+      (connectionStatus == ConnectionStatus.CELLULAR || connectionStatus == ConnectionStatus.LOCAL)
+    ) {
```

Comment:
Why not always force the status? The extra checks here don't quite make sense to me.

Reply:
We are supposed to prevent impossible scenarios from happening which is forcing WIFI or CELLULAR network when there is no actual network on the device.